test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusSerializerTests.cs

```diff
@@ -34,7 +34,15 @@ public void GaugeZeroDimension()
             using var meter = new Meter(Utils.GetCurrentMethodName());
             using var provider = Sdk.CreateMeterProviderBuilder()
                 .AddMeter(meter.Name)
-                .AddInMemoryExporter(metrics)
+                .AddInMemoryExporter(exportFunc: batch =>
+                {
+                    foreach (var metric in batch)
```

Comment:
Is there a way (e.g. list.AddRange) to reduce the boilerplate code (ideally keeping it a one liner)?

Reply:
I had that same thought while i was writing this. i'll spend some time on this today

Reply:
@reyang 
AddRange doesn't work here because there's not a built in way to convert Batch to IEnumerable.
If we need to convert Batch to something, I would prefer to convert it to a List.

My initial idea was to add a helper method to the Batch class itself.
Instead, my most recent change adds a helper method to only this test class. 
See: `private static List<Metric> BatchToList(Batch<Metric> batch)`

Reply:
Looks better, still have room for simplification. Could you make it even simpler?

Reply:
The simplest would be to add an additional Extension method that accepts `Action<T>`. I didn't think I could get away with adding both Func and Action to the PublicApi and between the two I chose Func.

Reply:
One simple way would be having an extension method just in this test project - `AddInMemoryExporter(some list type)` which handles the boilerplate job.

Reply:
I put a new extension method `AddInMemoryExporter(List<Metric>)` in `Tests.Shared` so it can be reused by multiple projects.
This helps reduce the number of changes in this PR, the individual Prometheus and OTLP unit tests don't need any changes.

Reply:
I don't think we want to encourage more use of this copy approach given the fact that Metric can be changed after the export.